ps_mbo.php

```diff
@@ -194,33 +230,60 @@ public function fetchModulesByController($ajax = false)
             return $data;
         }
 
-        return $this->context->smarty->fetch($this->template_dir . '/admin-end-content.tpl');
+        return $this->context->smarty->fetch($this->template_dir . '/include/admin-end-content-footer.tpl');
     }
-
-    public function hookDisplayAdminEndContent()
-    {
+    
+    protected function _handleAddonsConnectWithMbo() {
+        $return = false;
         if (Tools::getIsset('controller') && Tools::getValue('controller') == 'AdminPsMboModule') {
             $addonsConnect = $this->getAddonsConnectToolbar();
 
             $this->context->smarty->assign(array(
                 'addons_connect' => $addonsConnect,
             ));
 
-            return $this->context->smarty->fetch($this->template_dir . '/include/modal_addons_connect.tpl');
+            $return = $this->context->smarty->fetch($this->template_dir . '/include/modal_addons_connect.tpl');
         }
-
+        
+        return $return;
+    }
+    
+    protected function _handleTheme() {
+        $return = false;
+        
         $controller = Tools::getValue('controller');
         if ($controller == 'AdminThemes') {
             $this->context->smarty->assign(array(
                 'admin_module_ajax_url_psmbo' => $this->front_controller[0]
             ));
-            return $this->context->smarty->fetch($this->template_dir . '/admin-end-content-theme.tpl');
+            $return = $this->context->smarty->fetch($this->template_dir . '/admin-end-content-theme.tpl');
         }
+        
+        return $return;
+    }
 
+    public function hookDisplayAdminEndContent()
+    {
+        $connectWithMbo = $this->_handleAddonsConnectWithMbo();
+        if ($connectWithMbo !== false) {
+            return $connectWithMbo;
+        }
+        
+        $handleTheme = $this->_handleTheme();
+        if ($handleTheme !== false) {
+            return $handleTheme;
+        }
+        
         $content = '';
         $content .= $this->context->smarty->fetch($this->template_dir . '/modal.tpl');
-
-        if ($this->fetchModulesByController() !== false) {
+        
+        $controller_page = (Tools::getIsset('controller')) ? Tools::getValue('controller') : '';
+        $controllerWhiteList = array('AdminCarriers', 'AdminPayment');            
+        if (in_array($controller_page, $controllerWhiteList)) {
```

Comment:
```
if (!in_array($controller_page, $controllerWhiteList)) {
  return $content;
}
```
Alwasy return as fast as possible to prevent big or useless indentation.

Reply:
i don't think i can return faster because i still need to modal above to be called in the TPL